### PR TITLE
E2E Tests: Virustotal

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|-------------|
|   [ 3037 ](https://github.com/wazuh/wazuh-qa/issues/3037)        |

## Description
The test research was accomplished in https://github.com/wazuh/wazuh-qa/issues/2846. So, in this issue, we develop the E2E test to cover those use cases.

### Added

- Test to scanning a file using FIM and then do a request to Virustotal to check if it is malicious.
- Test to delete a malicious file using Active response and detected by Virustotal.

### Configuration options

To launch these tests, we must indicate the path of the inventory:

```
python -m pytest -vvvs tests/end_to_end/test_virustotal/ --inventory_path=/home/belen/Repositories/wazuh-qa/tests/end_to_end/general_playbooks/inventory.yml 
```
It should be noted that, for local environments, it will be necessary to create the inventory for the moment. It should contain the following:

```
all:
  hosts:
    wazuh-manager:
      ansible_connection: ssh
      ansible_user: vagrant
      ansible_ssh_private_key_file: <path_to_private_key>
      ansible_python_interpreter: /usr/bin/python3
     
    wazuh-agent:
      ansible_connection: ssh
      ansible_user: vagrant
      ansible_ssh_private_key_file: <path_to_private_key>
      ansible_python_interpreter: /usr/bin/python3

  vars:
    virustotal_key: VIRUSTOTAL-KEY

```

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |  tests/end_to_end/test_virustotal/          | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9047724/R1-tests-virustotal.2.zip)[🟢](https://github.com/wazuh/wazuh-qa/files/9047729/R2-tests-virustotal.2.zip)[🟢](https://github.com/wazuh/wazuh-qa/files/9047733/R3-tests-virustotal.2.zip) |  CentOS       |  040f8a4a2a56e8acd37235129e33afe4959bda7c       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but some changes are required

File: tests/end_to_end/test_virustotal/data/configuration/virustotal_key [hunk 1]
```diff
@@ -0,0 +1 @@
+3e4db70c621cd9dd9e3400254297eea03215987facca9931a42dcb86ffb8aa78
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_virustotal/data/configuration/virustotal_key]: Done in 040f8a4a2a56e8acd37235129e33afe4959bda7c

File: tests/end_to_end/test_virustotal/test_virustotal_active_response/data/playbooks/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      copy:
+        src: "{{ active_response_script }}"
+        dest: /var/ossec/active-response
+
+    - name: Install jq (CentOS)
+      become: true
+      shell: yum install jq
+      when: ansible_facts['distribution'] == "CentOS"
+
+    - name: Change remove threat owner and permissions
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_virustotal/test_virustotal_active_response/data/playbooks/configuration.yaml]: Done in 040f8a4a2a56e8acd37235129e33afe4959bda7c

File: tests/end_to_end/test_virustotal/test_virustotal_active_response/data/test_cases/cases_virustotal.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- name: Remove malicious file
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_virustotal/test_virustotal_active_response/data/test_cases/cases_virustotal.yaml]: Done in 040f8a4a2a56e8acd37235129e33afe4959bda7c

File: tests/end_to_end/test_virustotal/test_virustotal_fim/data/playbooks/configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  become: true
+  tasks:
+
+    - name: Delete folder
+      command: rm -r /test
+      ignore_errors: true
+
+    - name: Create folder
+      command: mkdir /test
+      ignore_errors: true
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_virustotal/test_virustotal_fim/data/playbooks/configuration.yaml]: Deleted command to remove folder in 040f8a4a2a56e8acd37235129e33afe4959bda7c

**Review by juliamagan — APPROVED** [review]

File: tests/end_to_end/test_virustotal/test_virustotal_fim/data/playbooks/configuration.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/end_to_end/test_virustotal/test_virustotal_fim/data/playbooks/configuration.yaml]: We will clean the environment in #3065

**Pull request merged**